python/cudf/cudf/core/column/string.py

```diff
@@ -97,6 +97,61 @@ def str_to_boolean(column: StringColumn):
     cudf.dtype("timedelta64[ns]"): str_cast.int2timedelta,
 }
 
+_NAN_INF_VARIATIONS = [
+    "nan",
+    "NAN",
+    "Nan",
+    "naN",
+    "nAN",
+    "NAn",
+    "-inf",
+    "-INF",
+    "-InF",
+    "-inF",
+    "-iNF",
+    "-INf",
+    "+inf",
+    "+INF",
+    "+InF",
+    "+inF",
+    "+iNF",
+    "+INf",
+    "+Inf",
+    "inf",
+    "INF",
+    "InF",
+    "inF",
+    "iNF",
+    "INf",
+]
```

Comment:
Would all of these really happen? 
If so, there are a couple of missing combinations: `nAn, -iNf, +iNf, iNf`
If not, would suggest perhaps only supporting `NaN, nan, NAN;  -Inf, -inf, -INF;  +Inf, +inf, +INF;  Inf, inf, INF`

Reply:
Maybe helpful: https://github.com/rapidsai/cudf/blob/2ef82f2983d6821141b6a7110b3fdecfd7e89e0f/python/cudf/cudf/core/tools/numeric.py#L243

Reply:
> Would all of these really happen? If so, there are a couple of missing combinations: `nAn, -iNf, +iNf, iNf` If not, would suggest perhaps only supporting `NaN, nan, NAN; -Inf, -inf, -INF; +Inf, +inf, +INF; Inf, inf, INF`

Thanks for catching this @davidwendt. The string to float conversion happens via NumPy machinery. FWIW, it appears that NumPy converts all strings to lower case equivalent and then performs the type-cast which makes it easier to handle these kinds of combinations, for ex:

```python
>>> s
array(['213', 'nan', 'NAN', 'INF'], dtype=object)
>>> s.astype('float')
array([213.,  nan,  nan,  inf])
```

I had these combinations added to match this behavior, if you think performing a `.lower` operation and then replacing is a better alternative we can switch to that as well.

Reply:
Ok. I think it is reasonable for libcudf to support all uppercase for this as documented for `std::stof` here: https://en.cppreference.com/w/cpp/string/basic_string/stof
This way, you could call `.upper()` which should be faster than `.replace()`.

This change is not difficult so let me know if you want me to open a PR for this.
Since 21.12 is now in burndown, we could also go ahead with this PR as is and make the change to libcudf and change cudf to use `.upper()` in the next release as well.

Reply:
> Since 21.12 is now in burndown, we could also go ahead with this PR as is and make the change to libcudf and change cudf to use `.upper()` in the next release as well.

This sounds good to me.

Reply:
> Maybe helpful:
> 
> https://github.com/rapidsai/cudf/blob/2ef82f2983d6821141b6a7110b3fdecfd7e89e0f/python/cudf/cudf/core/tools/numeric.py#L243

I think both these code-paths should be unified a little bit, since David will be adding the all uppercase support in next release, I've added a TODO to merge these two paths.